_source/_posts/2022-01-13-node-deploy-to-heroku.md

```diff
@@ -0,0 +1,492 @@
+
+To continue you will need:
+
+- [Visual Studio Code](https://code.visualstudio.com/)
+- [Node.Js](https://nodejs.org/en/)
+- An [Okta Developer Account](https://developer.okta.com/) (free forever, to handle your OAuth needs)
+- A [Heroku Account](https://signup.heroku.com/)
+- A [Git Repository](https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository)
+
+Of course, if you want to just see the code feel free [to view it on github](https://github.com/nickolasfisher/Okta_Heroku).
+
+## Create your Okta Application
+
+{% include setup/cli.md type="web" framework="node"
+   loginRedirectUri="http://localhost:3000/authorization-code/callback" signup="false" %}
+
+## Create your Express Application
+
+Now it's time to turn your attention to writing your application.   YOu will quickly scaffold your application by using the `express application generator` with the npx command `npx express-generator`.  Open the folder where your application will live and run the command `npx express-generator`.  
+
+### Install your Dependencies
+
+Next, run the command to install the default dependencies from express-generator.
+
+```console
+npm i
+```
+
+Finally, you will need to install the custom dependencies as well.  
+
+First, you will add `dotenv` to house your sensitive and environment-specific information.  
+
+```console
+npm dotenv@10.0.0
+```
+
+Next, you will want to add bootstrap.  YOu will use the bootstrap libraries in your `jade` templates once you begin writing the front end.  Bootstrap is a simple UI framework that has many samples to get you developing HTML pages quickly.
+
+```console
+npm bootstrap@5.1.3
+```
+
+You will need the `okta oidc-middleware` to help secure your application.  `express-session` is required when using the oidc-middleware so you will need to install that as well.
+
+```console
+npm install --save express-session@1.17.2
+npm install --save @okta/oidc-middleware@4.3.0
+```
+
+### Write your Server Code
+
+The express-generator package does a great job of scaffolding a simple Node.js application.  For this tutorial, you can leave most of the application the way it is.  You will need to make a few changes though.  
+
+First, add a new file called .env to the root of your application and add the following values.  These values can be found in `.okta.env`.
+
+```dotenv
+OKTA_OAUTH2_ISSUER={yourOktaDomain}/oauth2/default
+OKTA_OAUTH2_CLIENT_SECRET={yourClientSecret}
+OKTA_OAUTH2_CLIENT_ID={yourClientId}
+APP_BASEURL=http://localhost:3000
+```
+
+Next, open `app.js` and replace the code there with the following.
+
+```javascript
+require("dotenv").config();
```

Comment:
If you use the `.okta.env` file, you can use the following line of code instead of this:

```js
require('dotenv').config({ path: '.okta.env' })
```

Reply:
you'd also need to edit the values in that file.  see above.

Reply:
Please take a look at the express sample: https://github.com/okta-samples/okta-express-sample. We should follow a similar format for the sake of consistency.